charm/tests/integration/test_charm.py

```diff
@@ -36,3 +42,24 @@ async def test_build_and_deploy(ops_test: OpsTest):
         )
 
     assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
+
+
+async def test_tls(ops_test: OpsTest):
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "ch:self-signed-certificates",
+            application_name=ssc_app_name,
+            channel="edge",
+            trust=True,
+        ),
+    )
+    await asyncio.gather(ops_test.model.add_relation(APP_NAME, ssc_app_name))
+    await ops_test.model.wait_for_idle(apps=app_names, status="active")
+
+    address = await get_unit_address(ops_test, APP_NAME, 0)
+    url = f"https://{address}/"
+    try:
+        response = requests.get(url, verify=False)
+        assert response.status_code == 200
+    except requests.exceptions.RequestException:
+        assert False is True
```

Comment:
An assertion would fail the test, so we can reduce this to:
```suggestion
    response = requests.get(url, verify=False)
    assert response.status_code == 200
```

Or if we have a sensible error mesage,
```suggestion
    try:
        response = requests.get(url, verify=False)
        assert response.status_code == 200
    except requests.exceptions.RequestException as e:
        raise Exception("X went wrong") from e
```